Build fails for Python 3.11

### What I did:

$ cd Zope
$ tox -epy311

### What I expect to happen:

* Successful build.
* Caution: Running `tox -epy311` inside AccessControl fails because there is no `RestrictedPython` version supporting Python 3.11 (See https://github.com/zopefoundation/RestrictedPython/issues/222)

### What actually happened:

```
Getting distribution for 'AccessControl==5.3'.
  error: subprocess-exited-with-error

  × python setup.py bdist_wheel did not run successfully.
  │ exit code: 1
  ╰─> [201 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build/lib.macosx-12.0-x86_64-3.11
      creating build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/interfaces.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/User.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/Owned.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/ImplPython.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/Permission.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/AuthEncoding.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/users.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/SecurityManagement.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/PermissionMapping.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/logger_wrapper.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/PermissionRole.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/security.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/ImplC.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/__init__.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/ZopeGuards.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/safe_formatter.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/SpecialUsers.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/owner.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/Role.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/requestmethod.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/ZopeSecurityPolicy.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/SimpleObjectPolicies.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/unauthorized.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/class_init.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/userfolder.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/metaconfigure.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/Implementation.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/Permissions.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/tainted.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/rolemanager.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/SecurityManager.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/SecurityInfo.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      creating build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      copying src/AccessControl/tests/testSecurityManager.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      copying src/AccessControl/tests/actual_python.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      copying src/AccessControl/tests/test_userfolder.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      copying src/AccessControl/tests/test_requestmethod.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      copying src/AccessControl/tests/testModuleSecurity.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      copying src/AccessControl/tests/testImplementation.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      copying src/AccessControl/tests/__init__.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      copying src/AccessControl/tests/testZCML.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      copying src/AccessControl/tests/test_safeiter.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      copying src/AccessControl/tests/testZopeGuards.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      copying src/AccessControl/tests/test_rolemanager.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      copying src/AccessControl/tests/test_safe_formatter.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      copying src/AccessControl/tests/testPermissionRole.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      copying src/AccessControl/tests/test_users.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      copying src/AccessControl/tests/testPermissionMapping.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      copying src/AccessControl/tests/testClassSecurityInfo.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      copying src/AccessControl/tests/testZopeSecurityPolicy.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      copying src/AccessControl/tests/test_tainted.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      copying src/AccessControl/tests/test_owner.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      creating build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests/private_module
      copying src/AccessControl/tests/private_module/__init__.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests/private_module
      creating build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests/mixed_module
      copying src/AccessControl/tests/mixed_module/__init__.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests/mixed_module
      creating build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests/public_module
      copying src/AccessControl/tests/public_module/__init__.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests/public_module
      creating build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests/private_module/submodule
      copying src/AccessControl/tests/private_module/submodule/__init__.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests/private_module/submodule
      creating build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests/mixed_module/submodule
      copying src/AccessControl/tests/mixed_module/submodule/__init__.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests/mixed_module/submodule
      creating build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests/public_module/submodule
      copying src/AccessControl/tests/public_module/submodule/__init__.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests/public_module/submodule
      running egg_info
      writing src/AccessControl.egg-info/PKG-INFO
      writing dependency_links to src/AccessControl.egg-info/dependency_links.txt
      writing entry points to src/AccessControl.egg-info/entry_points.txt
      writing requirements to src/AccessControl.egg-info/requires.txt
      writing top-level names to src/AccessControl.egg-info/top_level.txt
      reading manifest file 'src/AccessControl.egg-info/SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      adding license file 'LICENSE.txt'
      writing manifest file 'src/AccessControl.egg-info/SOURCES.txt'
      copying src/AccessControl/_compat.h -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/cAccessControl.c -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/configure.zcml -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/meta.zcml -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/permissions.zcml -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      running build_ext
      building 'AccessControl.cAccessControl' extension
      creating build/temp.macosx-12.0-x86_64-3.11
      creating build/temp.macosx-12.0-x86_64-3.11/src
      creating build/temp.macosx-12.0-x86_64-3.11/src/AccessControl
      /usr/bin/clang -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -pipe -Os -Iinclude -Isrc -I/.../Zope/.tox/py311/include -I/.../3.11/include/python3.11 -c src/AccessControl/cAccessControl.c -o build/temp.macosx-12.0-x86_64-3.11/src/AccessControl/cAccessControl.o
      src/AccessControl/cAccessControl.c:428:2: warning: incompatible pointer to integer conversion initializing 'Py_ssize_t' (aka 'long') with an expression of type 'void *' [-Wint-conversion]
              NULL,                                   /* tp_print     */
              ^~~~
      /.../usr/include/sys/_types/_null.h:30:15: note: expanded from macro 'NULL'
      #define NULL  __DARWIN_NULL
                    ^~~~~~~~~~~~~
      /.../usr/include/sys/_types.h:52:23: note: expanded from macro '__DARWIN_NULL'
      #define __DARWIN_NULL ((void *)0)
                            ^~~~~~~~~~~
      src/AccessControl/cAccessControl.c:487:2: warning: incompatible pointer to integer conversion initializing 'Py_ssize_t' (aka 'long') with an expression of type 'void *' [-Wint-conversion]
              NULL,                                   /* tp_print     */
              ^~~~
      /.../usr/include/sys/_types/_null.h:30:15: note: expanded from macro 'NULL'
      #define NULL  __DARWIN_NULL
                    ^~~~~~~~~~~~~
      /.../usr/include/sys/_types.h:52:23: note: expanded from macro '__DARWIN_NULL'
      #define __DARWIN_NULL ((void *)0)
                            ^~~~~~~~~~~
      src/AccessControl/cAccessControl.c:546:2: warning: incompatible pointer to integer conversion initializing 'Py_ssize_t' (aka 'long') with an expression of type 'void *' [-Wint-conversion]
              NULL,                                   /* tp_print     */
              ^~~~
      /.../usr/include/sys/_types/_null.h:30:15: note: expanded from macro 'NULL'
      #define NULL  __DARWIN_NULL
                    ^~~~~~~~~~~~~
      /.../usr/include/sys/_types.h:52:23: note: expanded from macro '__DARWIN_NULL'
      #define __DARWIN_NULL ((void *)0)
                            ^~~~~~~~~~~
      src/AccessControl/cAccessControl.c:616:2: warning: incompatible pointer to integer conversion initializing 'Py_ssize_t' (aka 'long') with an expression of type 'void *' [-Wint-conversion]
              NULL,                                   /* tp_print     */
              ^~~~
      /.../usr/include/sys/_types/_null.h:30:15: note: expanded from macro 'NULL'
      #define NULL  __DARWIN_NULL
                    ^~~~~~~~~~~~~
      /.../usr/include/sys/_types.h:52:23: note: expanded from macro '__DARWIN_NULL'
      #define __DARWIN_NULL ((void *)0)
                            ^~~~~~~~~~~
      src/AccessControl/cAccessControl.c:1913:19: warning: '_PyUnicode_get_wstr_length' is deprecated [-Wdeprecated-declarations]
                    if (NATIVE_GET_SIZE(roles))
                        ^
      src/AccessControl/_compat.h:23:25: note: expanded from macro 'NATIVE_GET_SIZE'
      #define NATIVE_GET_SIZE PyUnicode_GET_SIZE
                              ^
      /.../3.11/include/python3.11/cpython/unicodeobject.h:248:7: note: expanded from macro 'PyUnicode_GET_SIZE'
            PyUnicode_WSTR_LENGTH(op) :                    \
            ^
      /.../3.11/include/python3.11/cpython/unicodeobject.h:442:35: note: expanded from macro 'PyUnicode_WSTR_LENGTH'
      #define PyUnicode_WSTR_LENGTH(op) _PyUnicode_get_wstr_length((PyObject*)op)
                                        ^
      /.../3.11/include/python3.11/cpython/unicodeobject.h:436:1: note: '_PyUnicode_get_wstr_length' has been explicitly marked deprecated here
      Py_DEPRECATED(3.3)
      ^
      /.../3.11/include/python3.11/pyport.h:381:54: note: expanded from macro 'Py_DEPRECATED'
      #define Py_DEPRECATED(VERSION_UNUSED) __attribute__((__deprecated__))
                                                           ^
      src/AccessControl/cAccessControl.c:1913:19: warning: 'PyUnicode_AsUnicode' is deprecated [-Wdeprecated-declarations]
                    if (NATIVE_GET_SIZE(roles))
                        ^
      src/AccessControl/_compat.h:23:25: note: expanded from macro 'NATIVE_GET_SIZE'
      #define NATIVE_GET_SIZE PyUnicode_GET_SIZE
                              ^
      /.../3.11/include/python3.11/cpython/unicodeobject.h:249:14: note: expanded from macro 'PyUnicode_GET_SIZE'
            ((void)PyUnicode_AsUnicode(_PyObject_CAST(op)),\
                   ^
      /.../3.11/include/python3.11/cpython/unicodeobject.h:571:1: note: 'PyUnicode_AsUnicode' has been explicitly marked deprecated here
      Py_DEPRECATED(3.3) PyAPI_FUNC(Py_UNICODE *) PyUnicode_AsUnicode(
      ^
      /.../3.11/include/python3.11/pyport.h:381:54: note: expanded from macro 'Py_DEPRECATED'
      #define Py_DEPRECATED(VERSION_UNUSED) __attribute__((__deprecated__))
                                                           ^
      src/AccessControl/cAccessControl.c:1913:19: warning: '_PyUnicode_get_wstr_length' is deprecated [-Wdeprecated-declarations]
                    if (NATIVE_GET_SIZE(roles))
                        ^
      src/AccessControl/_compat.h:23:25: note: expanded from macro 'NATIVE_GET_SIZE'
      #define NATIVE_GET_SIZE PyUnicode_GET_SIZE
                              ^
      /.../3.11/include/python3.11/cpython/unicodeobject.h:251:8: note: expanded from macro 'PyUnicode_GET_SIZE'
             PyUnicode_WSTR_LENGTH(op)))
             ^
      /.../3.11/include/python3.11/cpython/unicodeobject.h:442:35: note: expanded from macro 'PyUnicode_WSTR_LENGTH'
      #define PyUnicode_WSTR_LENGTH(op) _PyUnicode_get_wstr_length((PyObject*)op)
                                        ^
      /.../3.11/include/python3.11/cpython/unicodeobject.h:436:1: note: '_PyUnicode_get_wstr_length' has been explicitly marked deprecated here
      Py_DEPRECATED(3.3)
      ^
      /.../3.11/include/python3.11/pyport.h:381:54: note: expanded from macro 'Py_DEPRECATED'
      #define Py_DEPRECATED(VERSION_UNUSED) __attribute__((__deprecated__))
                                                           ^
      src/AccessControl/cAccessControl.c:1935:12: error: cannot combine with previous 'int' declaration specifier
                    int bool = PyObject_IsTrue(roles);
                        ^
      /.../usr/lib/clang/13.0.0/include/stdbool.h:15:14: note: expanded from macro 'bool'
      #define bool _Bool
                   ^
      src/AccessControl/cAccessControl.c:1935:17: error: expected identifier or '('
                    int bool = PyObject_IsTrue(roles);
                             ^
      src/AccessControl/cAccessControl.c:1936:12: error: expected expression
                    if (bool < 0)
                        ^
      /.../usr/lib/clang/13.0.0/include/stdbool.h:15:14: note: expanded from macro 'bool'
      #define bool _Bool
                   ^
      src/AccessControl/cAccessControl.c:1938:12: error: expected expression
                    if (bool)
                        ^
      /.../usr/lib/clang/13.0.0/include/stdbool.h:15:14: note: expanded from macro 'bool'
      #define bool _Bool
                   ^
      7 warnings and 4 errors generated.
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for AccessControl
  error: subprocess-exited-with-error

  × Running setup.py install for AccessControl did not run successfully.
  │ exit code: 1
  ╰─> [203 lines of output]
      running install
      /.../Zope/.tox/py311/lib/python3.11/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
        warnings.warn(
      running build
      running build_py
      creating build
      creating build/lib.macosx-12.0-x86_64-3.11
      creating build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/interfaces.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/User.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/Owned.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/ImplPython.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/Permission.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/AuthEncoding.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/users.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/SecurityManagement.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/PermissionMapping.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/logger_wrapper.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/PermissionRole.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/security.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/ImplC.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/__init__.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/ZopeGuards.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/safe_formatter.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/SpecialUsers.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/owner.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/Role.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/requestmethod.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/ZopeSecurityPolicy.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/SimpleObjectPolicies.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/unauthorized.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/class_init.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/userfolder.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/metaconfigure.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/Implementation.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/Permissions.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/tainted.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/rolemanager.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/SecurityManager.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/SecurityInfo.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      creating build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      copying src/AccessControl/tests/testSecurityManager.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      copying src/AccessControl/tests/actual_python.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      copying src/AccessControl/tests/test_userfolder.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      copying src/AccessControl/tests/test_requestmethod.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      copying src/AccessControl/tests/testModuleSecurity.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      copying src/AccessControl/tests/testImplementation.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      copying src/AccessControl/tests/__init__.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      copying src/AccessControl/tests/testZCML.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      copying src/AccessControl/tests/test_safeiter.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      copying src/AccessControl/tests/testZopeGuards.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      copying src/AccessControl/tests/test_rolemanager.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      copying src/AccessControl/tests/test_safe_formatter.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      copying src/AccessControl/tests/testPermissionRole.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      copying src/AccessControl/tests/test_users.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      copying src/AccessControl/tests/testPermissionMapping.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      copying src/AccessControl/tests/testClassSecurityInfo.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      copying src/AccessControl/tests/testZopeSecurityPolicy.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      copying src/AccessControl/tests/test_tainted.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      copying src/AccessControl/tests/test_owner.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests
      creating build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests/private_module
      copying src/AccessControl/tests/private_module/__init__.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests/private_module
      creating build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests/mixed_module
      copying src/AccessControl/tests/mixed_module/__init__.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests/mixed_module
      creating build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests/public_module
      copying src/AccessControl/tests/public_module/__init__.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests/public_module
      creating build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests/private_module/submodule
      copying src/AccessControl/tests/private_module/submodule/__init__.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests/private_module/submodule
      creating build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests/mixed_module/submodule
      copying src/AccessControl/tests/mixed_module/submodule/__init__.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests/mixed_module/submodule
      creating build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests/public_module/submodule
      copying src/AccessControl/tests/public_module/submodule/__init__.py -> build/lib.macosx-12.0-x86_64-3.11/AccessControl/tests/public_module/submodule
      running egg_info
      writing src/AccessControl.egg-info/PKG-INFO
      writing dependency_links to src/AccessControl.egg-info/dependency_links.txt
      writing entry points to src/AccessControl.egg-info/entry_points.txt
      writing requirements to src/AccessControl.egg-info/requires.txt
      writing top-level names to src/AccessControl.egg-info/top_level.txt
      reading manifest file 'src/AccessControl.egg-info/SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      adding license file 'LICENSE.txt'
      writing manifest file 'src/AccessControl.egg-info/SOURCES.txt'
      copying src/AccessControl/_compat.h -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/cAccessControl.c -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/configure.zcml -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/meta.zcml -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      copying src/AccessControl/permissions.zcml -> build/lib.macosx-12.0-x86_64-3.11/AccessControl
      running build_ext
      building 'AccessControl.cAccessControl' extension
      creating build/temp.macosx-12.0-x86_64-3.11
      creating build/temp.macosx-12.0-x86_64-3.11/src
      creating build/temp.macosx-12.0-x86_64-3.11/src/AccessControl
      /usr/bin/clang -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -pipe -Os -Iinclude -Isrc -I/.../Zope/.tox/py311/include -I/.../3.11/include/python3.11 -c src/AccessControl/cAccessControl.c -o build/temp.macosx-12.0-x86_64-3.11/src/AccessControl/cAccessControl.o
      src/AccessControl/cAccessControl.c:428:2: warning: incompatible pointer to integer conversion initializing 'Py_ssize_t' (aka 'long') with an expression of type 'void *' [-Wint-conversion]
              NULL,                                   /* tp_print     */
              ^~~~
      /.../usr/include/sys/_types/_null.h:30:15: note: expanded from macro 'NULL'
      #define NULL  __DARWIN_NULL
                    ^~~~~~~~~~~~~
      /.../usr/include/sys/_types.h:52:23: note: expanded from macro '__DARWIN_NULL'
      #define __DARWIN_NULL ((void *)0)
                            ^~~~~~~~~~~
      src/AccessControl/cAccessControl.c:487:2: warning: incompatible pointer to integer conversion initializing 'Py_ssize_t' (aka 'long') with an expression of type 'void *' [-Wint-conversion]
              NULL,                                   /* tp_print     */
              ^~~~
      /.../usr/include/sys/_types/_null.h:30:15: note: expanded from macro 'NULL'
      #define NULL  __DARWIN_NULL
                    ^~~~~~~~~~~~~
      /.../usr/include/sys/_types.h:52:23: note: expanded from macro '__DARWIN_NULL'
      #define __DARWIN_NULL ((void *)0)
                            ^~~~~~~~~~~
      src/AccessControl/cAccessControl.c:546:2: warning: incompatible pointer to integer conversion initializing 'Py_ssize_t' (aka 'long') with an expression of type 'void *' [-Wint-conversion]
              NULL,                                   /* tp_print     */
              ^~~~
      /.../usr/include/sys/_types/_null.h:30:15: note: expanded from macro 'NULL'
      #define NULL  __DARWIN_NULL
                    ^~~~~~~~~~~~~
      /.../usr/include/sys/_types.h:52:23: note: expanded from macro '__DARWIN_NULL'
      #define __DARWIN_NULL ((void *)0)
                            ^~~~~~~~~~~
      src/AccessControl/cAccessControl.c:616:2: warning: incompatible pointer to integer conversion initializing 'Py_ssize_t' (aka 'long') with an expression of type 'void *' [-Wint-conversion]
              NULL,                                   /* tp_print     */
              ^~~~
      /.../usr/include/sys/_types/_null.h:30:15: note: expanded from macro 'NULL'
      #define NULL  __DARWIN_NULL
                    ^~~~~~~~~~~~~
      /.../usr/include/sys/_types.h:52:23: note: expanded from macro '__DARWIN_NULL'
      #define __DARWIN_NULL ((void *)0)
                            ^~~~~~~~~~~
      src/AccessControl/cAccessControl.c:1913:19: warning: '_PyUnicode_get_wstr_length' is deprecated [-Wdeprecated-declarations]
                    if (NATIVE_GET_SIZE(roles))
                        ^
      src/AccessControl/_compat.h:23:25: note: expanded from macro 'NATIVE_GET_SIZE'
      #define NATIVE_GET_SIZE PyUnicode_GET_SIZE
                              ^
      /.../3.11/include/python3.11/cpython/unicodeobject.h:248:7: note: expanded from macro 'PyUnicode_GET_SIZE'
            PyUnicode_WSTR_LENGTH(op) :                    \
            ^
      /.../3.11/include/python3.11/cpython/unicodeobject.h:442:35: note: expanded from macro 'PyUnicode_WSTR_LENGTH'
      #define PyUnicode_WSTR_LENGTH(op) _PyUnicode_get_wstr_length((PyObject*)op)
                                        ^
      /.../3.11/include/python3.11/cpython/unicodeobject.h:436:1: note: '_PyUnicode_get_wstr_length' has been explicitly marked deprecated here
      Py_DEPRECATED(3.3)
      ^
      /.../3.11/include/python3.11/pyport.h:381:54: note: expanded from macro 'Py_DEPRECATED'
      #define Py_DEPRECATED(VERSION_UNUSED) __attribute__((__deprecated__))
                                                           ^
      src/AccessControl/cAccessControl.c:1913:19: warning: 'PyUnicode_AsUnicode' is deprecated [-Wdeprecated-declarations]
                    if (NATIVE_GET_SIZE(roles))
                        ^
      src/AccessControl/_compat.h:23:25: note: expanded from macro 'NATIVE_GET_SIZE'
      #define NATIVE_GET_SIZE PyUnicode_GET_SIZE
                              ^
      /.../3.11/include/python3.11/cpython/unicodeobject.h:249:14: note: expanded from macro 'PyUnicode_GET_SIZE'
            ((void)PyUnicode_AsUnicode(_PyObject_CAST(op)),\
                   ^
      /.../3.11/include/python3.11/cpython/unicodeobject.h:571:1: note: 'PyUnicode_AsUnicode' has been explicitly marked deprecated here
      Py_DEPRECATED(3.3) PyAPI_FUNC(Py_UNICODE *) PyUnicode_AsUnicode(
      ^
      /.../3.11/include/python3.11/pyport.h:381:54: note: expanded from macro 'Py_DEPRECATED'
      #define Py_DEPRECATED(VERSION_UNUSED) __attribute__((__deprecated__))
                                                           ^
      src/AccessControl/cAccessControl.c:1913:19: warning: '_PyUnicode_get_wstr_length' is deprecated [-Wdeprecated-declarations]
                    if (NATIVE_GET_SIZE(roles))
                        ^
      src/AccessControl/_compat.h:23:25: note: expanded from macro 'NATIVE_GET_SIZE'
      #define NATIVE_GET_SIZE PyUnicode_GET_SIZE
                              ^
      /.../3.11/include/python3.11/cpython/unicodeobject.h:251:8: note: expanded from macro 'PyUnicode_GET_SIZE'
             PyUnicode_WSTR_LENGTH(op)))
             ^
      /.../3.11/include/python3.11/cpython/unicodeobject.h:442:35: note: expanded from macro 'PyUnicode_WSTR_LENGTH'
      #define PyUnicode_WSTR_LENGTH(op) _PyUnicode_get_wstr_length((PyObject*)op)
                                        ^
      /.../3.11/include/python3.11/cpython/unicodeobject.h:436:1: note: '_PyUnicode_get_wstr_length' has been explicitly marked deprecated here
      Py_DEPRECATED(3.3)
      ^
      /.../3.11/include/python3.11/pyport.h:381:54: note: expanded from macro 'Py_DEPRECATED'
      #define Py_DEPRECATED(VERSION_UNUSED) __attribute__((__deprecated__))
                                                           ^
      src/AccessControl/cAccessControl.c:1935:12: error: cannot combine with previous 'int' declaration specifier
                    int bool = PyObject_IsTrue(roles);
                        ^
      /.../usr/lib/clang/13.0.0/include/stdbool.h:15:14: note: expanded from macro 'bool'
      #define bool _Bool
                   ^
      src/AccessControl/cAccessControl.c:1935:17: error: expected identifier or '('
                    int bool = PyObject_IsTrue(roles);
                             ^
      src/AccessControl/cAccessControl.c:1936:12: error: expected expression
                    if (bool < 0)
                        ^
      /.../usr/lib/clang/13.0.0/include/stdbool.h:15:14: note: expanded from macro 'bool'
      #define bool _Bool
                   ^
      src/AccessControl/cAccessControl.c:1938:12: error: expected expression
                    if (bool)
                        ^
      /.../usr/lib/clang/13.0.0/include/stdbool.h:15:14: note: expanded from macro 'bool'
      #define bool _Bool
                   ^
      7 warnings and 4 errors generated.
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> AccessControl
```

### What version of Python and Zope/Addons I am using:

* Zope `master`
* Python 3.11.0a5

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -235,7 +235,7 @@ jobs:
           # when we ask it to load tests from that directory. This
           # might also save some build time?
           unzip -n dist/AccessControl-*whl -d src
-          pip install -U -e .[test]
+          pip install --pre -U -e .[test]
       - name: Run tests with C extensions
         if: ${{ !startsWith(matrix.python-version, 'pypy') }}
         run: |

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -102,6 +102,7 @@ jobs:
           - "3.8"
           - "3.9"
           - "3.10"
+          - "3.11.0-alpha.6"
         os: [ubuntu-20.04, macos-latest]
         exclude:
           - os: macos-latest
@@ -166,6 +167,7 @@ jobs:
           && startsWith(github.ref, 'refs/tags')
           && startsWith(runner.os, 'Mac')
           && !startsWith(matrix.python-version, 'pypy')
+          && !startsWith(matrix.python-version, '3.11.0-alpha.6')
         env:
           TWINE_PASSWORD: ${{ secrets.TWINE_PASSWORD }}
         run: |
@@ -185,6 +187,7 @@ jobs:
           - "3.8"
           - "3.9"
           - "3.10"
+          - "3.11.0-alpha.6"
         os: [ubuntu-20.04, macos-latest]
         exclude:
           - os: macos-latest
@@ -224,7 +227,9 @@ jobs:
       - name: Install AccessControl
         run: |
           pip install -U wheel setuptools
-          pip install -U coverage
+          # coverage has a wheel on PyPI for a future python version which is
+          # not ABI compatible with the current one, so build it from sdist:
+          pip install -U --no-binary :all: coverage
           pip install -U 'faulthandler; python_version == "2.7" and platform_python_implementation == "CPython"'
           # Unzip into src/ so that testrunner can find the .so files
           # when we ask it to load tests from that directory. This

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -147,7 +147,7 @@ jobs:
           python setup.py bdist_wheel
           # Also install it, so that we get dependencies in the (pip) cache.
           pip install -U 'faulthandler; python_version == "2.7" and platform_python_implementation == "CPython"'
-          pip install .[test]
+          pip install --pre .[test]
 
       - name: Check AccessControl build
         run: |

--- a/.gitignore
+++ b/.gitignore
@@ -28,4 +28,5 @@ lib64
 log/
 parts/
 pyvenv.cfg
+testing.log
 var/

--- a/.manylinux-install.sh
+++ b/.manylinux-install.sh
@@ -31,6 +31,7 @@ for PYBIN in /opt/python/*/bin; do
     if \
        [[ "${PYBIN}" == *"cp27"* ]] || \
        [[ "${PYBIN}" == *"cp35"* ]] || \
+       [[ "${PYBIN}" == *"cp311"* ]] || \
        [[ "${PYBIN}" == *"cp36"* ]] || \
        [[ "${PYBIN}" == *"cp37"* ]] || \
        [[ "${PYBIN}" == *"cp38"* ]] || \
@@ -39,10 +40,10 @@ for PYBIN in /opt/python/*/bin; do
         "${PYBIN}/pip" install -e /io/
         "${PYBIN}/pip" wheel /io/ -w wheelhouse/
         if [ `uname -m` == 'aarch64' ]; then
-         cd /io/
-         "${PYBIN}/pip" install tox
-         "${PYBIN}/tox" -e py
-         cd ..
+          cd /io/
+          "${PYBIN}/pip" install tox
+          "${PYBIN}/tox" -e py
+          cd ..
         fi
         rm -rf /io/build /io/*.egg-info
     fi

--- a/.meta.toml
+++ b/.meta.toml
@@ -2,7 +2,7 @@
 # https://github.com/zopefoundation/meta/tree/master/config/c-code
 [meta]
 template = "c-code"
-commit-id = "3a251aab94eff9cf2aa2fd87d7266593251e4b76"
+commit-id = "e3088883ccf38fe07dded689981c6e748a45300f"
 
 [python]
 with-appveyor = true

--- a/.meta.toml
+++ b/.meta.toml
@@ -2,13 +2,13 @@
 # https://github.com/zopefoundation/meta/tree/master/config/c-code
 [meta]
 template = "c-code"
-commit-id = "121e74bd9c9718abd9a1a079e6ede252c1a0ba7d"
+commit-id = "3a251aab94eff9cf2aa2fd87d7266593251e4b76"
 
 [python]
 with-appveyor = true
 with-windows = false
 with-pypy = false
-with-future-python = false
+with-future-python = true
 with-legacy-python = true
 with-sphinx-doctests = false
 

--- a/.meta.toml
+++ b/.meta.toml
@@ -2,7 +2,7 @@
 # https://github.com/zopefoundation/meta/tree/master/config/c-code
 [meta]
 template = "c-code"
-commit-id = "3489c7048afe51c20264785ea8abfca5ef234771"
+commit-id = "eadba64bcfb4aca29fffa2fd06689ea445b8c710"
 
 [python]
 with-appveyor = true

--- a/.meta.toml
+++ b/.meta.toml
@@ -2,7 +2,7 @@
 # https://github.com/zopefoundation/meta/tree/master/config/c-code
 [meta]
 template = "c-code"
-commit-id = "e3088883ccf38fe07dded689981c6e748a45300f"
+commit-id = "3489c7048afe51c20264785ea8abfca5ef234771"
 
 [python]
 with-appveyor = true

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -6,7 +6,7 @@ For changes before version 3.0, see ``HISTORY.rst``.
 5.4 (unreleased)
 ----------------
 
-- Nothing changed yet.
+- Add support for Python 3.11 (as of 3.11.0a6).
 
 
 5.3.1 (2022-03-29)

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -24,6 +24,10 @@ environment:
     - python: 39-x64
     - python: 310
     - python: 310-x64
+    # `multibuild` cannot install non-final versions as they are not on
+    # ftp.python.org, so we skip Python 3.11 until its final release:
+    # - python: 311
+    # - python: 311-x64
 
 install:
   - "SET PYTHONVERSION=%PYTHON%"

--- a/setup.cfg
+++ b/setup.cfg
@@ -19,3 +19,14 @@ per-file-ignores =
 ignore =
     .editorconfig
     .meta.toml
+
+[isort]
+force_single_line = True
+combine_as_imports = True
+sections = FUTURE,STDLIB,THIRDPARTY,ZOPE,FIRSTPARTY,LOCALFOLDER
+known_third_party = six, docutils, pkg_resources
+known_zope =
+known_first_party =
+default_section = ZOPE
+line_length = 79
+lines_after_imports = 2

--- a/src/AccessControl/cAccessControl.c
+++ b/src/AccessControl/cAccessControl.c
@@ -1932,10 +1932,10 @@ c_rolesForPermissionOn(PyObject *perm, PyObject *object,
           */
           else 
   	    {
-	      int bool = PyObject_IsTrue(roles);
-	      if (bool < 0)
+	      int roles_is_true = PyObject_IsTrue(roles);
+	      if (roles_is_true < 0)
 		  goto end;
-	      if (bool)
+	      if (roles_is_true)
 	        {
 		    PyObject *list_roles = PySequence_List(roles);
 		    Py_DECREF(roles);

--- a/tox.ini
+++ b/tox.ini
@@ -11,10 +11,12 @@ envlist =
     py38,py38-pure
     py39,py39-pure
     py310,py310-pure
+    py311,py311-pure
     coverage
 
 [testenv]
 usedevelop = true
+pip_pre = true
 deps =
 setenv =
     pure: PURE_PYTHON=1
@@ -41,12 +43,22 @@ commands =
 [testenv:lint]
 basepython = python3
 skip_install = true
-deps =
-    flake8
-    check-manifest
-    check-python-versions >= 0.19.1
-    wheel
 commands =
+    isort --check-only --diff {toxinidir}/src {toxinidir}/setup.py
     flake8 src setup.py
     check-manifest
     check-python-versions
+deps =
+    check-manifest
+    check-python-versions >= 0.19.1
+    wheel
+    flake8
+    isort
+
+[testenv:isort-apply]
+basepython = python3
+commands_pre =
+deps =
+    isort
+commands =
+    isort {toxinidir}/src {toxinidir}/setup.py []

